core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -385,6 +368,141 @@ sealed abstract class MultiSigSigner
 
 object MultiSigSigner extends MultiSigSigner
 
+/** Used to sign a [[org.bitcoins.core.protocol.script.P2SHScriptPubKey]] */
+sealed abstract class P2SHSigner extends BitcoinSigner[P2SHSpendingInfo] {
+  override def sign(
+      spendingInfo: UTXOSpendingInfo,
+      unsignedTx: Transaction,
+      isDummySignature: Boolean,
+      spendingInfoToSatisfy: P2SHSpendingInfo)(
+      implicit ec: ExecutionContext): Future[TxSigComponent] = {
+    if (spendingInfoToSatisfy != spendingInfo) {
+      Future.fromTry(TxBuilderError.WrongSigner)
+    } else {
+      val (_, output, inputIndex, _) =
+        relevantInfo(spendingInfo, unsignedTx)
+
+      (spendingInfoToSatisfy, spendingInfoToSatisfy.redeemScript) match {
+        case (_: P2SHNestedSegwitV0UTXOSpendingInfo,
+              p2wpkh: P2WPKHWitnessSPKV0) =>
+          val witTx = WitnessTransaction.toWitnessTx(unsignedTx)
+          val p2shScriptSig = P2SHScriptSignature(p2wpkh)
+          val oldInput = witTx.inputs(inputIndex.toInt)
+          val updatedInput = TransactionInput(oldInput.previousOutput,
+                                              p2shScriptSig,
+                                              oldInput.sequence)
+
+          val uwtx = {
+            val u = unsignedTx.updateInput(inputIndex.toInt, updatedInput)
+            WitnessTransaction.toWitnessTx(u)
+          }
+
+          val wtxSigComp = WitnessTxSigComponentP2SH(transaction = uwtx,
+                                                     inputIndex = inputIndex,
+                                                     output = output,
+                                                     flags =
+                                                       Policy.standardFlags)
+
+          val hashForSig = TransactionSignatureSerializer.hashForSignature(
+            txSigComponent = wtxSigComp,
+            hashType = spendingInfo.hashType)
+
+          //sign the hash
+          val signature: ECDigitalSignature = {
+            if (isDummySignature) {
+              EmptyDigitalSignature
+            } else {
+              val sig = spendingInfo.signers.head.sign(hashForSig.bytes)
```

Comment:
Hmm... this seems wrong. the `doSign()` method takes care of this everywhere else. Can't we just call that here?